clang/test/Sema/attr-nolock2.cpp

```diff
@@ -0,0 +1,84 @@
+// RUN: %clang_cc1 -fsyntax-only -fblocks -verify %s
```

Comment:
Nit: I’d either merge this test w/ the other `attr-nolock` ones, or give it a more descriptive name than ‘attr-nolock2’.

Reply:
Thanks, yes, I will merge them (other than the one that is checking the AST dump).

Reply:
Hm, the standard diagnostic for diagnosing incompatible attributes produces an error, and the last-pass constraint analysis is skipped when there are errors. I'm going to keep those constraint analysis tests in a separate file and give the files more descriptive names, e.g. -parsing, -sema, -constraints.

Reply:
It’s usually not required to separate code that errors and code that is supposed to compile without errors, but if it is required for something like this, then sure, go ahead.

>  -parsing, -sema, -constraints.

Well, *parsing*-related tests should probably go in `test/Parser` (from what I’ve seen so far, you also haven’t made any changes in the parser, though).

Reply:
OK, maybe "syntax" then...